### PR TITLE
backport: merge bitcoin-core/gui#337, #354, #317, #349, #403, #418, #416, #449, #477, #563, #554, #615 (qt backports: part 2)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

| `develop` (3a18f087)                                       | This PR                                                      |
| ------------------------------------------------------------ | ------------------------------------------------------------ |
| image | image |
| image | image |

## Breaking changes

None observed.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas (note: N/A)
- [x] I have added or updated relevant unit/integration/functional/e2e tests (note: N/A)
- [x] I have made corresponding changes to the documentation (note: N/A)
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
utACK 1662905dfbb2ba0b05faa4b2c8bae987b9a358ee

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/qt/forms/optionsdialog.ui [hunk 1]
```diff
@@ -291,13 +291,23 @@
...
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
+       <item>
+        <widget class="QCheckBox" name="enableServer">
+         <property name="toolTip">
+          <string extracomment="Tooltip text for Options window setting that enables the RPC server.">This allows you or a third party tool to communicate with the node through command-line and JSON-RPC commands.</string>
```
Comment (knst) [inline src/qt/forms/optionsdialog.ui:304]: gui-416: let's have release notes for it?

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -1678,9 +1679,6 @@ void BitcoinGUI::changeEvent(QEvent *e)
 #endif
     if (e->type() == QEvent::StyleChange) {
         updateNetworkState();
-#ifdef ENABLE_WALLET
-        updateWalletStatus();
-#endif
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/qt/bitcoingui.cpp:1683]: `updateWalletStatus` is changed into an `assert` ([source](https://github.com/bitcoin-core/gui/pull/403/files#diff-2ecf8cbf369cf3d2f3d2b1cf5cfe4c1a647d63e11e2885d2fd0ac11fb5f7a804R1343-R1344)) and without it being removed, the client crashes (see below) 

```
$ gdb ./src/qt/dash-qt
[...]
(gdb) r
Starting program: /home/eclair/Repositories/dashpay/dash/src/qt/dash-qt
[...]
dash-qt: qt/bitcoingui.cpp:1915: void BitcoinGUI::updateWalletStatus(): Assertion `walletFrame' failed.

Thread 1 "dash-qt" received signal SIGABRT, Aborted.
0x00007ffff78a9e3c in ?? () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) bt
#0  0x00007ffff78a9e3c in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff785afb2 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff7845472 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff7845395 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x00007ffff7853eb2 in __assert_fail () from /lib/x86_64-linux-gnu/libc.so.6
#5  0x00005555556f06a5 in BitcoinGUI::updateWalletStatus (this=0x555558c7fcb0) at qt/bitcoingui.cpp:1915
#6  0x00005555556f256f in BitcoinGUI::changeEvent (this=0x555558c7fcb0, e=0x7fffffffd560) at qt/bitcoingui.cpp:1683
#7  0x0000555556c99908 in QWidget::event(QEvent*) ()
#8  0x0000555556d5e4d8 in QMainWindow::event(QEvent*) ()
#9  0x0000555556c61313 in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
#10 0x0000555556c6ae08 in QApplication::notify(QObject*, QEvent*) ()
#11 0x0000555556999238 in QCoreApplication::sendEvent(QObject*, QEvent*) ()
#12 0x0000555556c677c0 in QApplication::setStyle(QStyle*) [clone .part.0] ()
#13 0x000055555570986e in GUIUtil::loadStyleSheet (fForceUpdate=<optimized out>) at qt/guiutil.cpp:1016
#14 0x0000555555710f37 in GUIUtil::loadTheme (fForce=151) at qt/guiutil.cpp:1602
#15 0x00005555556e3a65 in BitcoinGUI::BitcoinGUI (this=0x555558c7fcb0, node=..., networkStyle=0x5555583b97e0,
    parent=<optimized out>) at qt/bitcoingui.cpp:91
#16 0x00005555556d4d38 in BitcoinApplication::createWindow (this=0x7fffffffd8d0, networkStyle=0x5555583b97e0)
    at qt/bitcoin.cpp:322
#17 0x00005555556da0ed in GuiMain (argc=<optimized out>, argv=<optimized out>) at qt/bitcoin.cpp:811
#18 0x00007ffff784624a in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#19 0x00007ffff7846305 in __libc_start_main () from /lib/x86_64-linux-gnu/libc.so.6
#20 0x00005555556d2d8e in _start ()
```

File: src/qt/bitcoingui.cpp [hunk 3]
```diff
@@ -112,14 +112,16 @@ BitcoinGUI::BitcoinGUI(interfaces::Node& node, const NetworkStyle* networkStyle,
...
         connect(walletFrame, &WalletFrame::createWalletButtonClicked, [this] {
             auto activity = new CreateWalletActivity(getWalletController(), this);
             connect(activity, &CreateWalletActivity::finished, activity, &QObject::deleteLater);
             activity->create();
         });
+        connect(walletFrame, &WalletFrame::message, [this](const QString& title, const QString& message, unsigned int style) {
+            this->message(title, message, style);
+        });
+        connect(walletFrame, &WalletFrame::currentWalletSet, [this] { updateWalletStatus(); });
+        setCentralWidget(walletFrame);
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/qt/bitcoingui.cpp]: Resolved in latest push!

File: src/qt/rpcconsole.cpp [hunk 4]
```diff
@@ -962,15 +962,19 @@ void RPCConsole::clear(bool keep_prompt)
...
+#endif
     ui->messagesWidget->document()->setDefaultStyleSheet(
         QString(
                 "table { }"
                 "td.time { " + GUIUtil::getThemedStyleQString(GUIUtil::ThemedStyle::TS_SECONDARY) + " font-size: %1pt; } "
                 "td.message { " + GUIUtil::getThemedStyleQString(GUIUtil::ThemedStyle::TS_PRIMARY) + " font-size: %1pt; white-space:pre-wrap; } "
                 "td.cmd-request, b { " + GUIUtil::getThemedStyleQString(GUIUtil::ThemedStyle::TS_COMMAND) + " } "
                 "td.cmd-error, .secwarning { " + GUIUtil::getThemedStyleQString(GUIUtil::ThemedStyle::TS_ERROR) + " }"
-            ).arg(consoleFontSize)
+            ).arg(QString("%1pt").arg(consoleFontSize))
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/qt/rpcconsole.cpp]: Resolved in latest push!

File: src/qt/forms/optionsdialog.ui (hunk 1 above)
Comment (kwvg) [reply src/qt/forms/optionsdialog.ui:304]: Resolved in latest push!

**Review by knst — APPROVED** [review]
utACK 15bd92a9a7725d2944e52c432a69042f51e652e1

File: src/qt/bitcoingui.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/qt/bitcoingui.cpp:1683]: hmmm... let's maybe wrap it in `if` then?
```suggestion
#ifdef ENABLE_WALLET
        if (walletFrame) {
            updateWalletStatus();
        }
#endif
```

[the remark this answers is not in the document]
Comment (kwvg) [reply src/qt/bitcoingui.cpp]: Resolved in latest push!

**Review by UdjinM6 — APPROVED** [review]
utACK 1a2c02a42ddcf8f2025f2260e64791f0142eaa5f

**Pull request merged**